advanced_authoring/02_lesson.rst

```diff
@@ -0,0 +1,244 @@
+          }
+    )
+
+Here we name the @Action ``Stub``. After that we declare some outputs in an
+array using the ``@Output`` annotation and we provide a string for each output's
+name. This name will be used in the CloudSlang operation to retrieve the
+output's value. Finally, in another array we declare the responses that our
+action can return. (These are not actually used by CloudSlang operations.)
+
+Each @Response is made of several parts:
+
+  - ``text``: name of the response
+  - ``field``: output to be checked
+  - ``value``: value to check against
+  - ``matchType``: type of check
+    (from com.hp.oo.sdk.content.plugin.ActionMetadata.MatchType)
+  - ``responseType``: type of response
+    (from com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType)
+  - ``isDefault``: whether or not response is the default response
+  - ``isOnFail``: whether or not response is the failure response
+
+@Action Method
+--------------
+
+Now we'll write the method that will be called by the CloudSlang operation.
+By convention, we name this method ``execute``.The method must conform to the
+following signature:
+
+.. code-block:: java
+
+  public Map<String, String> methodName(@Param(name) Type param, ...)
+
+The map that is returned contains the names and values of the outputs of the
+action. The names are the same ones that were declared using ``@Output`` in the
+``@Action`` annotation. The parameters that are declared with the ``@Param``
+annotation are the inputs that the method will receive from the CloudSlang
+operation. Each ``@Param`` will contain at least value for its name. It may also
+contain a ``required`` boolean to override the default of ``false``.
+
+Let's create our method and add a couple of parameters:
+
+.. code-block:: java
+
+  public Map<String, String> execute(@Param(value = "text", required = true) String text,
+                                     @Param("forceFail") String forceFail){
+  }
+
+Inside the method we'll just write a few lines of code that will allow us to get
+an idea of how CloudSlang communicates with the action method and test it out a
+bit.
+
+.. code-block:: java
+
+  Map<String, String> results = new HashMap<>();
+  results.put("message", "Hello from Java. The text you sent me is: " + text);
+  if(Boolean.parseBoolean(forceFail)){
+      results.put("returnCode", "-1");
+  }
+  else{
+      results.put("returnCode", "0");
+  }
+  return results;
+
+We start by creating a map that will eventually contain the output values of the
+method (by convention, we call this map ``results``). Then we add a values for
+the two outputs and return the map.
+
+Tests
+-----
+
+Next, let's set up some tests. In the **src\\test\\java** folder let's create a new
+package named ``io.cloudslang.tutorial.httpclient.actions``. And within that
+package, create a new class named ``HttpClientActionTest``.
```

Comment:
why name it HttpClientActionTest and HttpClientAction if the action doesn't do anything with HTTP?